LiskHQ/lisk-sdk
# Update BFT to use state store - Closes #4316

- Add forger info table and move maxHeightPreviouslyForged to this table
- Rename chain meta to chain state table
- Update source code of BFT to use state store to update data

### What was the problem?
- BFT was not consistently using state store
- max height previously forged was stored in the chain state, but it has to be in different table

### How did I solve it?
- Create forger info table so that it can be handled in different way when snapshotting / restoring
- Update BFT to use state store

### How to manually test it?
- All the test should pass
- Observe database info that forger_info is populated
- Observe finality is saved

### Review checklist

- [ ] The PR resolves #4316
